packages/apm/src/hubextensions.ts

```diff
@@ -55,15 +56,17 @@ function startSpan(spanOrSpanContext?: Span | SpanContext, forceNoChild: boolean
     span = new Span(spanOrSpanContext, that);
   }
 
-  if (span.sampled === undefined && span.transaction !== undefined) {
+  // We only roll the dice on sampling for "root" spans (transactions) because the childs inherit this state
+  if (span.sampled === undefined && !span.isChildSpan()) {
```

Comment:
Suggestion for readability:

```js
if (span.isTransaction() && span.sampled === undefined) {
```

Then we can probably remove the comment from above, or say

```js
// Decide whether this transaction should be sent to Sentry. Child spans inherit this value.
```

Which makes me think that if we had two different types, `Span` and `Transaction`, `Span.sampled()` would read from `Transaction.sampled` instead of keeping a value internally.

Just again thinking about misuse...

```js
let t = new Span({transaction: "test"});
let child = t.child({sampled: true});      // this is semantically "wrong", but the value will be silently ignored
```